### PR TITLE
CPSRE-1015: migrate coral off of CircleCI and move to GHA

### PR DESCRIPTION
**Pull request by iambrianchan**
## What does this PR do?

This PR moves the `build-test` and `build-test-deploy` steps off CircleCI and onto Github Actions.

## These changes will impact:

- [ ] commenters
- [ ] moderators
- [ ] admins
- [x] developers

## How do I test this PR?
Make sure that `build-test` is working, and I have already tried to mimic the `build-test-deploy` step by simulating using a tag `v9.9.9`

### CODE REVIEW

**Comment by nick-funk** [comment]
All hail @iambrianchan, bringer of GitHub greatness to the dark lands of Coral! 😄 

This looks awesome and thank you so much for helping us move to GitHub actions, this is incredibly exciting and wonderful to see!

🙌  🙌  🙌

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        name: Define SHORT_SHA with commit short sha
+        run: echo "SHORT_SHA=`echo ${GITHUB_SHA} | cut -c1-8`" >> $GITHUB_ENV
+      -
+        name: Setup Node14.x
+        uses: actions/setup-node@v3
+        with:
+          node-version: '14.x'
+      -
+        name: Install npm 8
+        run: npm i -g npm@8.0.0 --registry=https://registry.npmjs.org
```
Comment (nick-funk) [inline .github/workflows/build-and-test.yml:40]: +1, thank you!

File: .github/workflows/build-and-test.yml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+      -
+        name: npm Lockfile Version
+        uses: mansona/npm-lockfile-version@v1
+        with:
+          version: 2
```
Comment (nick-funk) [inline .github/workflows/build-and-test.yml:45]: this is neat, didn't know GHA had options for this in the flow!

**Review by nick-funk — APPROVED** [review]

**Review by existere — APPROVED** [review]
:shipit:

**Pull request merged**